loopy/schedule/tree.py

```diff
@@ -50,7 +50,9 @@
 NodeT = TypeVar("NodeT", bound=Hashable)
 
 
-@dataclass(frozen=True)
+# Not frozen because it is slower. Tree objects are immutable, and offer no
+# way to mutate the tree.
+@dataclass(frozen=False)
```

Comment:
```suggestion
@dataclass(frozen=__debug__)
```
(And adapt the comment suitably?)

Reply:
https://github.com/inducer/pytools/pull/272